tcblib/slack_functions.py

```diff
@@ -0,0 +1,52 @@
+import boto3
+import hmac
+import slack
+
+def validate_request(req):
+    """Validate slack request to prevent the hax."""
+    slack_signing_secret = boto3.client('ssm').get_parameter(
+        Name='/slack/tcb/signing_secret',
+        WithDecryption=True
+    )['Parameter']['Value']
+    signing_secret = str.encode(slack_signing_secret)
+
+    slack_request_timestamp = req['headers']['X-Slack-Request-Timestamp']
+    request_body = req['body']
+
+    token = 'v0:{0}:{1}'.format(
+        slack_request_timestamp,
+        request_body
+    )
+    encoded_token = str.encode(token)
```

Comment:
is this still necessary?

Reply:
yis ._.

Reply:
Are you sure? What is `str.encode()` actually doing to `token`?

Reply:
When I remove it, the hmac thing on [line 24](https://github.com/evagonz/tcb/pull/1/files/d2eafe866b3169f406b3b3ae1ef3dd4c0a7aad7c#diff-f2d790781c1b678c54a5f7246cf38dc9R24) fails because it's not encoded. The-mule does this too, just in [a separate function.](https://github.com/civisanalytics/the-mule/blob/master/themule/mulelib/request_validator.py#L123)